File: Web_Vue.Server/ViewModels/Auth/CaptchaViewModel.cs
namespace Web_Vue.Server.ViewModels.Auth;

/// <summary> 登入圖形驗證碼 ViewModel </summary>
public class CaptchaViewModel
{
    /// <summary> Challenge ID（呼叫登入 API 時需一併帶入） </summary>
    public string ChallengeId { get; set; } = string.Empty;

    /// <summary> SVG 驗證碼圖片 data URI (data:image/svg+xml;base64,...) </summary>
    public string ImageBase64 { get; set; } = string.Empty;

    /// <summary> 過期時間 (UTC) </summary>
    public DateTime ExpiresAt { get; set; }
}

File: web_vue.client/types/viewmodel/Auth/CaptchaViewModel.ts
// !! 此檔案由 generate-models.mjs 自動生成，請勿手動修改 !!
// 來源：Web_Vue.Server/ViewModels
/** 登入圖形驗證碼 ViewModel */
export interface CaptchaViewModel {
  /** Challenge ID（呼叫登入 API 時需一併帶入） */
  ChallengeId: string
  /** SVG 驗證碼圖片 data URI (data:image/svg+xml;base64,...) */
  ImageBase64: string
  /** 過期時間 (UTC) */
  ExpiresAt: string
}
